crates/re_types_builder/src/codegen/python.rs

```diff
@@ -399,16 +546,27 @@ fn code_for_struct(
         }
 
         // init override handling
-        let init_override_name = format!("{}__init_override", obj.snake_case_name());
-        let (init_define_arg, init_func) = if overrides.contains(&init_override_name) {
-            ("init=False".to_owned(), init_override_name.clone())
+        let old_init_override_name = format!("{}__init_override", obj.snake_case_name());
+        let init_define_arg = if ext_class.has_init || overrides.contains(&old_init_override_name) {
+            "init=False".to_owned()
```

Comment:
`if` `has_init`, then `init=False`… I don't follow

Reply:
Adding comment to clarify:
```
        // If the `ExtensionClass` has its own `__init__` then we need to pass the `init=False` argument
        // to the `@define` decorator. Otherwise the generated `__init__` from `@define` will take
        // precedence.
```